### PR TITLE
#184 Fixed team validation and activating update button

### PR DESCRIPTION
**Pull request by EugeneValko**

### CODE REVIEW

File: frontend/src/app/modules/settings/team/edit-team/edit-team.component.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { Subscription } from 'rxjs';
 export class EditTeamComponent extends BaseComponent implements OnDestroy {
     private id: number | undefined;
 
-    public team: ITeam;
+    public team: ITeam = {} as ITeam;
```
Comment (Erbidium) [inline frontend/src/app/modules/settings/team/edit-team/edit-team.component.ts]: Why is such initialization needed?

File: frontend/src/app/modules/settings/team/edit-team/edit-team.component.ts [hunk 2]
```diff
@@ -51,7 +51,15 @@ export class EditTeamComponent extends BaseComponent implements OnDestroy {
                 .pipe(this.untilThis)
                 .subscribe({
                     next: (team) => {
-                        this.team = team;
```
Comment (Erbidium) [inline frontend/src/app/modules/settings/team/edit-team/edit-team.component.ts:54]: Why separate fields intialization is needed?

**Review by Erbidium — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/settings/team/edit-team/edit-team.component.ts (hunk 2 above)
Comment (EugeneValko) [reply frontend/src/app/modules/settings/team/edit-team/edit-team.component.ts:54]: because the TeamPreferencesComponent did not receive the current Team

**Review by Erbidium — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]
LGTM

**Pull request merged**